### ISSUE TITLE
Experiment with ServerSideRender

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/oik-blocks#16]
Gutenberg provides a mechanism called `ServerSideRender` which allows a block to be rendered by the server. It’s intended for legacy code, which is exactly what some of the shortcodes are. We should experiment with this solution to attempt to render the real block that’s displayed for a subset of shortcodes. 
Starting with:

Block | Shortcode | Consideration
-- | -- | --
Oik-block/wp | wp | attributes: v, m, p
oik-block/geshi | bw_geshi | Generic Syntax Highlighting for lang=none,css,php,html,javascript,jquery
Oik-block/search | bw_search 
Oik-block/login | bw_login | consider protected content
Oik-block/list | bw_list | many attributes!
Oik-block/googlemap | bw_show_googlemap | needs to cater for Google’s JavaScript

For some logic the server side rendering may require the global $post to be set.
Other logic needs to know the current user.

**Comment by bobbingwide** [comment]
See #12 for a summary of required blocks.

**Comment by bobbingwide** [comment]
While developing the Fields block I discovered that the version of `ServerSideRender` that passes the `post_id` to the REST API is in `wp.editor`;  not `wp.components` as currently documented. 

The code was changed in PR 8720, but there don't appear to be any deprecation notices. 
Each block developed so far should be updated. 

Perhaps there should be a summary of where to import components from.
